core/drivers/crypto/hisilicon/hisi_qm.c

```diff
@@ -143,49 +147,127 @@ static void qm_db(struct hisi_qm *qm, uint16_t qn, uint8_t cmd, uint16_t index,
 	io_write64(qm->io_base + QM_DOORBELL_SQ_CQ_BASE, doorbell);
 }
 
+static void qm_mb_write(struct hisi_qm *qm, void *src)
+{
+	uintptr_t dst = qm->io_base + QM_MAILBOX_BASE;
+	unsigned long tmp0 = 0, tmp1 = 0;
+
+	/* 128bits should be written to hardware at one time */
+	asm volatile ("ldp %0, %1, %3\n"
```

Comment:
The `ldp` operation can be done in C code and the `stp` operation is already available as `write_64bit_pair()`.